pkg/roachprod/vm/gce/dns.go

```diff
@@ -323,6 +343,27 @@ func (n *dnsProvider) listSRVRecords(
 	return records, nil
 }
 
+// lockRecordByName locks the record with the given name and returns a function
+// that can be used to unlock it. The lock is used to prevent concurrent
+// operations on the same record.
+func (n *dnsProvider) lockRecordByName(name string) func() {
```

Comment:
Nit: this API could also be written as:

```go
func (n *dnsProvider) withRecordLock(name string, f func() error) error {
    // lock
    // defer unlock()

    return f()
}
```

Mostly because you already changed the callers anyway into closures to allow a `defer unlock` pattern. With the API above it's slightly more ergonomic as callers don't need to get a handle to the lock every time and then `defer` the unlock themselves.

This only makes sense if we are planning to only lock once based on name, which seems to be the case here.

Current approach is also fine btw, just throwing this idea out there.

Reply:
Makes sense. Done.